### ISSUE TITLE
DMD linker failure when building plugins on OS X

### ISSUE THREAD

**Issue by robsaunders** [AuburnSounds/Dplug#190]
looks like the examples aren't compiling... I waited a few weeks to see if this was just a transitory thing but it seems they are broken long term perhaps?

I compile the release tool with:
```
dub build --build=debug --arch=x86_64 --compiler=dmd
```

This works, however, the distort plugin for example, upon running:
```
drelease --compiler dmd -v
```

I get the output:
```
<snip>
ld: GOT load reloc does not point to a movq instruction in '_D3std4conv22__T10emplaceRefTaTaTaZ10emplaceRefFNaNbNiNfKaaZv' from .dub/build/VST-debug-posix.osx-x86_64-dmd_2075-68BAAF666205FB21359CFA04DB02FBDF/distort.o for architecture x86_64
```

I can provide full verbose output if required, but it basically just doesn't seem to be linking correctly. This applies for both the VST, AU configs, in release or debug builds.

`simple_mono_synth` works fine, so I'm guessing it's perhaps something to do with linking to cocoa? In the exports file, _dplugAUCarbonViewEntryPoint is listed, and it's worth bringing up that for 64-bit vsts, carbon is not supported, does this have anything to do with it?

Any pointers would be helpful... I'm somewhat new to D and am already feeling a lot of the problems commonly associated with it.

**Comment by p0nce** [comment]
`ld: GOT load reloc` is a DMD regression that I have not reported since about a year. I just lack the time to reduce this case. I was waiting for DMD to fix it itself but didn't happen either :)
You can use ldc instead (it comes with a DUB too) on OSX.

**Comment by p0nce** [comment]
TODO
- [ ] Report back the DMD linker regression when building on OS X a plugin

**Comment by robsaunders** [comment]
Update: it works with LDC, so it's purely DMD breaking. If I can help out in some way let me know.

**Comment by p0nce** [comment]
The only way forward is to open an issue in: https://issues.dlang.org/
With a minimal test case (implying no DUB or release tool, just a DMD command-line and 1 file).

What I usually do is copy the used Dplug files into the same directory as the test case, use path-based dependencies in dub.json and then use:
`dub --combined -v` to get the DMD command-line.

If you can isolate the problem it would be much appreciated..
The process of getting to that minimal test case may take up to 4 hours though...

**Comment by p0nce** [comment]
Might be https://issues.dlang.org/show_bug.cgi?id=15342

**Comment by p0nce** [comment]
Fixed in 6cabdb056da9937381c6094fdfdb95443e2970b6
